core/trace/tracer.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package trace
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+)
+
+const (
+	// ErrTracerDying is used to indicate to *third parties* that the
+	// tracer worker is dying, instead of catacomb.ErrDying, which is
+	// unsuitable for propagating inter-worker.
+	// This error indicates to consuming workers that their dependency has
+	// become unmet and a restart by the dependency engine is imminent.
+	ErrTracerDying = errors.ConstError("tracer worker is dying")
```

Comment:
Is this the right place for this to live? Considering it pertains to the functionality of the worker.